gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -504,12 +505,15 @@ public void handleLaunchFlowEvent(DagActionStore.DagAction action) {
       if (optionalJobExecutionPlanDag.isPresent()) {
         addDag(optionalJobExecutionPlanDag.get(), true, true);
       }
```

Comment:
sorry that I'm forgetting... where do we log/handle the case of no exec plan because the `FlowSpec` was invalid?  is that done by the caller?

Reply:
the method `FlowCompilationValidationHelper.createExecutionPlanIfValid` does all the necessary checks -> whether or not it can compile, if there's an ongoing execution and concurrent executions are allowed, and that the plan is not null/empty. If any of the failure conditions are met, the return value is `Optional.absent()`